modules/light-clients/08-wasm/keeper/msg_server_test.go

```diff
@@ -79,20 +88,176 @@ func (suite *KeeperTestSuite) TestMsgStoreCode() {
 				// Verify events
 				expectedEvents := sdk.Events{
 					sdk.NewEvent(
-						"store_wasm_code",
+						types.EventTypeStoreWasmCode,
 						sdk.NewAttribute(types.AttributeKeyWasmCodeHash, hex.EncodeToString(res.Checksum)),
 					),
+					sdk.NewEvent(
+						sdk.EventTypeMessage,
+						sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+					),
 				}
 
 				for _, evt := range expectedEvents {
 					suite.Require().Contains(events, evt)
 				}
 			} else {
-				suite.Require().Error(err)
 				suite.Require().ErrorIs(err, tc.expError)
 				suite.Require().Nil(res)
 				suite.Require().Empty(events)
 			}
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestMsgMigrateContract() {
+	oldCodeHash := sha256.Sum256(wasmtesting.Code)
+
+	newByteCode := []byte("MockByteCode-TestMsgMigrateContract")
+
+	govAcc := authtypes.NewModuleAddress(govtypes.ModuleName).String()
+
+	var (
+		newCodeHash    []byte
+		msg            *types.MsgMigrateContract
+		expClientState *types.ClientState
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, newCodeHash, []byte("{}"))
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					data, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					return &wasmvmtypes.Response{Data: data}, wasmtesting.DefaultGasUsed, nil
+				}
+			},
+			nil,
+		},
+		{
+			"success: update state",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, newCodeHash, []byte("{}"))
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					// the code hash written in here will be overwritten
+					expClientState = types.NewClientState([]byte{1}, []byte(ibctesting.InvalidID), clienttypes.NewHeight(2000, 2))
```

Comment:
I personally found it a bit misleading that you are using `InvalidID` here as the code hash and then realised that you are changing it to the actually elected code hash...

Reply:
Yes but should I change it?

Reply:
I would change it, just on the principle of least surprise :) If I was initially confused, I guess another engineer might as well be, and it's nice if we can avoid that. Happy to change this my self, if you want!